cargo-mode.el

```diff
@@ -145,7 +145,7 @@ If PROMPT is non-nil, modifies the command."
                               buffer-file-name
                               (string-prefix-p project-root (file-truename buffer-file-name)))))
     (setq cargo-mode--last-command (list name cmd project-root))
-    (compile cmd t)
+    (compile cmd nil)
```

Comment:
comint mode is used intentionally to be able to answer interactive prompts if there are any

will this work now?

Reply:
Right, do you know some example I can test this with? 

Perhaps it would be best to have this configurable per command. wdyt?

Reply:
for example `cargo login`

why is it a problem to use comint for every command?

Reply:
Entering of token for `cargo login` is not working with this change.

Problem with comint for every command is that none of the `compilation-mode` commands works and buffer is not read-only. For example:

 - when pressing `n` for `next-error-no-select` it enters letter `n` into the comint buffer 
 - `TAB` is indenting the content of comint buffer instead of running `compilation-next-error`
 -  and so on.

Perhaps `defcustom` list of compilation-mode friendly cargo command like `build`, `check`, `clippy`, `run` etc. can solve this problem? And people will be able to customize it further.

Reply:
maybe we can add an option that toggle comint mode. it can be enabled by default

(if some-option (compile cmd t) (compile cmd nil))